lib/sensors/accelerometer.hpp

```diff
@@ -24,10 +24,10 @@ constexpr std::uint8_t kAlternativeAccelerometerAddress = 0x18;
 class Accelerometer : public II2cMuxSensor<core::RawAccelerationData> {
  public:
   static std::optional<Accelerometer> create(core::ILogger &logger,
-                                             std::shared_ptr<io::II2c> i2c,
+                                             const std::shared_ptr<io::II2c> &i2c,
                                              const std::uint8_t channel,
                                              const std::uint8_t device_address);
-  ~Accelerometer();
+  ~Accelerometer() override = default;
```

Comment:
How does this work and why isn't it in the cpp file anymore?

Reply:
default destructor is the same as giving it a body that is just `{}`. I moved it to the header cos it might as well be there tbh, not much point in having an `~Accelerometer() = default` entry in the cpp

I should check what the difference between declaring it at all or not is

Reply:
declaring things `default` does nothing unless we're overloading constructors - which we aren't. I've removed them all except in the i2cmux template where it is needed to compile